subsys/pm/pm.c

```diff
@@ -68,6 +68,18 @@ static inline void pm_state_notify(bool entering_state)
 	k_spin_unlock(&pm_notifier_lock, pm_notifier_key);
 }
 
+static inline int32_t ticks_expiring_sooner(int32_t ticks1, int32_t ticks2)
+{
+	/*
```

Comment:
This function assumes the only allowed negative value of the parameters is -1 expressing "FOREVER".
I would add run-time check in debug code:
```
__ASSERT(ticks1 >= -1, "ticks1 has unexpected negative value");
__ASSERT(ticks2 >= -1, "ticks2 has unexpected negative value");
...
```